falcon/routing/static.py

```diff
@@ -33,13 +36,17 @@ class StaticRoute(object):
     # minimizes how much can be included in the payload.
     _MAX_NON_PREFIXED_LEN = 512
 
-    def __init__(self, prefix, directory, downloadable=False):
+    def __init__(self, prefix, directory, downloadable=False, fallback_filename=None):
         if not prefix.startswith('/'):
             raise ValueError("prefix must start with '/'")
 
         if not os.path.isabs(directory):
             raise ValueError('directory must be an absolute path')
 
+        fallback_path = os.path.join(directory, fallback_filename or '')
+        if fallback_filename is not None and not os.path.isfile(fallback_path):
+            raise ValueError('fallback_filename does not exists')
+
```

Comment:
Suggestion: It may simplify things a bit and remove the need for L59 altogether if `self._fallback_filename` were assigned directly above, e.g.:

```py
if fallback_filename is None:
    self._fallback_filename = None
else:
    self._fallback_filename = os.path.join(directory, fallback_filename)
    if not os.path.isfile(self._fallback_filename):
        raise ValueError('fallback_filename does not exists')
```

Reply:
I wanted to keep the same style: parameter check then assignment to instance variable.
I agree that it's cleaner this way

I'll change it